inc/dbutils.class.php

```diff
@@ -909,57 +909,83 @@ public function getTreeLeafValueName($table, $ID, $withcomment = false, $transla
       $name    = "";
       $comment = "";
 
-      $SELECTNAME    = "`$table`.`name`, '' AS transname";
-      $SELECTCOMMENT = "`$table`.`comment`, '' AS transcomment";
-      $JOIN          = '';
+      $SELECTNAME    = new \QueryExpression("'' AS ". $DB->quoteName('transname'));
+      $SELECTCOMMENT = new \QueryExpression("'' AS " . $DB->quoteName('transcomment'));
+      $JOIN          = [];
+      $JOINS         = [];
       if ($translate) {
          if (Session::haveTranslations($this->getItemTypeForTable($table), 'name')) {
-            $SELECTNAME  = "`$table`.`name`, `namet`.`value` AS transname";
-            $JOIN       .= " LEFT JOIN `glpi_dropdowntranslations` AS namet
-                              ON (`namet`.`itemtype` = '".$this->getItemTypeForTable($table)."'
-                                 AND `namet`.`items_id` = `$table`.`id`
-                                 AND `namet`.`language` = '".$_SESSION['glpilanguage']."'
-                                 AND `namet`.`field` = 'name')";
+            $SELECTNAME = new \QueryExpression("namet.value AS ". $DB->quoteName('transname'));
+            $JOINS['glpi_dropdowntranslations AS namet'] = [
+               'ON' => [
+                  'namet'  => 'items_id',
+                  $table   => 'id', [
+                     'AND' => [
+                        'namet.itemtype'  => $this->getItemTypeForTable($table),
+                        'namet.language'  => $_SESSION['glpilanguage'],
+                        'namet.field'     => 'name'
+                     ]
+                  ]
+               ]
+            ];
          }
          if (Session::haveTranslations($this->getItemTypeForTable($table), 'comment')) {
-            $SELECTCOMMENT  = "`$table`.`comment`, `namec`.`value` AS transcomment";
-            $JOIN          .= " LEFT JOIN `glpi_dropdowntranslations` AS namet
-                              ON (`namec`.`itemtype` = '".$this->getItemTypeForTable($table)."'
-                                 AND `namec`.`items_id` = `$table`.`id`
-                                 AND `namec`.`language` = '".$_SESSION['glpilanguage']."'
-                                 AND `namec`.`field` = 'comment')";
+            $SELECTCOMMENT = new \QueryExpression("namec.value AS " . $DB->quoteName('transcomment'));
```

Comment:
I think the usage of `QueryExpression` is useless here.
```suggestion
            $SELECTCOMMENT = 'namec.value AS transcomment';
```